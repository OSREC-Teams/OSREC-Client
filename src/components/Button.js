import styled from 'styled-components';

export default styled.button`
  background-color: #6c3b91;
  border-radius: 0.25rem;
  border: 2px solid;
  border-color: #6c3b91;
  padding: ${props => (props.padding ? props.padding : '0.375rem 0.75rem')};
  font-size: 1.5rem;
  font-weight: 900;
  text-align: center;
  color: white;
  transition: 0.2s;

  &:hover {
    background-color: #8449b2;
    border-color: #8449b2;
    cursor: pointer;
  }

  &:active {
    background-color: #593177;
    border-color: #593177;
  }

  &:focus {
    outline: 0;
  }
`;
