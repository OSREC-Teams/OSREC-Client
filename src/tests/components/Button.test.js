import React from 'react';
import Button from 'components/Button';
import { shallow } from 'enzyme';

test('Button renders an html button', () => {
  const button = shallow(<Button>Beautiful Button</Button>);
  expect(button).toMatchSnapshot();
});
