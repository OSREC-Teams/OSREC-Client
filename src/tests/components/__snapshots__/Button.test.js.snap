// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders an html button 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  background-color: #6c3b91;
  border-radius: 0.25rem;
  border: 2px solid;
  border-color: #6c3b91;
  padding: ",
          [Function],
          ";
  font-size: ",
          [Function],
          ";
  font-weight: 900;
  text-align: center;
  color: white;
  transition: 0.2s;

  &:hover {
    background-color: #8449b2;
    border-color: #8449b2;
    cursor: pointer;
  }

  &:active {
    background-color: #593177;
    border-color: #593177;
  }

  &:focus {
    outline: 0;
  }
",
        ],
      },
      "displayName": "styled.button",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Beautiful Button
</StyledComponent>
`;
